skxray/cdi.py

```diff
@@ -119,9 +125,339 @@ def convolution(array1, array2):
     `this issue on github <https://github.com/Nikea/scikit-xray/issues/258>`_
     for details.
     """
-    fft_norm = lambda x:  np.fft.fftshift(np.fft.fftn(x)) / np.sqrt(np.size(x))
-    fft_1 = fft_norm(array1)
-    fft_2 = fft_norm(array2)
-    return np.abs(np.fft.ifftshift(np.fft.ifftn(fft_1*fft_2)) *
-                  np.sqrt(np.size(array2)))
+    fft_1 = _fft_helper(array1) / np.sqrt(np.size(array1))
+    fft_2 = _fft_helper(array2) / np.sqrt(np.size(array2))
+    return np.abs(_ifft_helper(fft_1*fft_2) * np.sqrt(np.size(array2)))
+
+
+def pi_modulus(array_in, diff_array,
+               thresh_v=1e-12):
+    """
+    Transfer sample from real space to q space.
+    Use constraint based on diffraction pattern from experiments.
+
+    Parameters
+    ----------
+    array_in : array
```

Comment:
This probably also needs a better name like 'proposed_phase'.  It also make more sense to me for the measured data (`diff_array` or what ever it gets renamed to) to be the first input